File: go/bunsan/pm/repository.go
package pm

import "time"

//go:generate bunsan-mockgen -gofile=$GOFILE

type Repository interface {
	Create(sourcePath string, strip bool) error
	CreateRecursively(rootPath string, strip bool) error
	Extract(pkg, destination string) error
	Install(pkg, destination string) error
	ForceUpdate(pkg, destination string) error
	Update(pkg, destination string, lifetime time.Time) error
	NeedUpdate(pkg, destination string, lifetime time.Time) (bool, error)
	CleanCache() error
	Close() error
}

File: bunsan/pm_net/Bunsan.PM/Repository.cs
using System;

using Bunsan.PM.Detail;

namespace Bunsan.PM
{
	public class Repository
	{
		private String ConfigPath;

		public Repository (String configPath)
		{
			ConfigPath = configPath;
		}

		public void Create (String path, bool strip)
		{
			RepositoryImplementation.Create (ConfigPath, path, strip);
		}

		public void Create (String path)
		{
			RepositoryImplementation.Create (ConfigPath, path);
		}

		public void Clean ()
		{
			RepositoryImplementation.Clean (ConfigPath);
		}

		public void Extract (String package, String path)
		{
			RepositoryImplementation.Extract (ConfigPath, package, path);
		}
	}
}
